### ISSUE TITLE
Handling 3D data like time evolution of 2D fluid flow.

### ISSUE THREAD

**Issue by hammermm** [lu-group/deeponet-fno#2]
Thank you Lu for such exciting work.

RIght now, I want to use DeepONet on flow problems. My input is of the dimension (Samples, Height, Width, Time) or images corresponding to different time steps depicting the evolution of flow. Like my input is velocity from. t = 1 to m time steps and I want to predict the next n time steps [ batch, :,:,:m] -> [ batch,:,:, m: m+n].

How should I preprocess the data for such a problem? Is there any available code implementation for such flow problem using deepOnet?

Also, can we use DeepOnet like FNO-2D( more of RNN type structure)?

**Comment by lululxvi** [comment]
Yes, you can use DeepONet. There are different possible ways for input and output. The easiest way is using time step t as the input, and time step t+1 as the output. Or you can use multiple time steps as the input.
